### DISCUSSION TITLE
contributing/development/compiling/cross-compiling_for_ios_on_linux

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#284] [User-contributed notes]
# contributing/development/compiling/cross-compiling_for_ios_on_linux

The procedure for this is somewhat complex and requires a lot of steps, but once you have the environment properly configured you can compile Godot for iOS anytime you want. Disclaimer: While it is...

https://docs.godotengine.org/en/stable/contributing/development/compiling/cross-compiling_for_ios_on_linux.html

**Comment by hardaker** [comment]
FYI, navigating the apple pages to find the right actual download link for xcode is hard and keeps changing.  Here's the latest place to grab the needed file as of 2024-12-18: https://developer.apple.com/download/all/

**Comment by acarlstein** [comment]
I tried to automate the installation of the components, `check_ios_cross_compile_requirements.sh` version `1.0.0`, and run the steps of this documentation, `configure.sh` version `0.0.5`.

Here is the (link)[https://github.com/godotengine/godot-docs/issues/10964] to the issue with the bash shell code I was working on, in case someone is interested and want to contribute.

**Reply by acarlstein** [reply]
Updated the `configure.sh` to version `0.0..6` which will create the file `build_ios_godot.sh` so the user don't have to bother remembering the last step if they want to.

**Reply by acarlstein** [reply]
[link to issue](https://github.com/godotengine/godot-docs/issues/10964)

**Reply by acarlstein** [reply]
Something I'm missing or doing wrong because the last step, I get:
```
scons: *** No SConstruct file found.
File "/usr/lib/python3/dist-packages/SCons/Script/Main.py", line 965, in _main
```

**Comment by not-magic** [comment]
Is this cross-compiling possible with the current apple SDKs? Unless I can't find the right download, the current Xcode 16.4 no longer has the C++ sdk headers

**Reply by hardaker** [reply]
I never got it functional....  Thought I came close a few times, but nope.
